test/minikube/001-kube.bats

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bats
+#
+# Tests of podman kube commands with minikube
+#
+
+load ../system/helpers.bash
+
+function setup(){
+    # only set up the minikube cluster before the first test
+    if [ "$BATS_TEST_NUMBER" -eq 1 ]; then
+        minikube start
+        echo $output >&2
+    fi
+    # wait for a bit so all the services in the cluster come up
+    sleep 10
+}
+
+###############################################################################
+# BEGIN tests
+
+@test "minikube - check cluster is up" {
+    run minikube kubectl get nodes
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"Ready"* ]]
+    run minikube kubectl get pods
+    [ "$status" -eq 0 ]
+    [[ "$output" == "No resources found in default namespace." ]]
+}
+
+@test "minikube - deploy generated container yaml to minikube" {
+    cname="test-ctr"
+    fname="/tmp/f$(random_string 6).yaml"
+    run_podman container create --name $cname $IMAGE top
+    [ "$status" -eq 0 ]
+    run_podman kube generate -f $fname $cname
+    [ "$status" -eq 0 ]
+
+    # deploy to the minikube cluster
+    run minikube kubectl -- apply -f $fname
+    [ "$status" -eq 0 ]
+    [[ "$output" == "pod/$cname-pod created" ]]
+    # need to wait for the pod to be created and started before checking status
+    sleep 5
+    run minikube kubectl get pods
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"Running"* ]]
+
+}
+
+@test "minikube - deploy generated pod yaml to minikube" {
+    pname="test-pod"
+    cname1="test-ctr1"
+    cname2="test-ctr2"
+    fname="/tmp/f$(random_string 6).yaml"
+
+    run_podman pod create --name $pname --publish 9999:8888
+    [ "$status" -eq 0 ]
+    run_podman container create --name $cname1 --pod $pname $IMAGE sleep 1000
+    [ "$status" -eq 0 ]
+    run_podman container create --name $cname2 --pod $pname $IMAGE sleep 2000
+    [ "$status" -eq 0 ]
+    run_podman kube generate -f $fname $pname
+    [ "$status" -eq 0 ]
+
+    #deploy to the minikube cluster
+    run minikube kubectl -- apply -f $fname
+    [ "$status" -eq 0 ]
```

Comment:
Even in this use, `run` other than `run_podman`, this sort of test is horrible. A better thing to do is
```
run minikube ...
assert "$status" -eq 0 "exit status of minikube etc"
```

Reply:
Fixed for status, but I am not seeing a way to user `assert` to check for a substring in the output.

Reply:
```console
$ ack ' assert .*=~' test/system
```
(or `grep -R`, or whatever your favorite search tool is)

Reply:
thanks, fixed